### PR TITLE
feat(config): add EXTERNAL_PORT to decouple server port from URL generation

### LINKED ISSUE #66: Decouple server listen port from public-facing Swagger URL port
### Problem

`PORT` in the config serves double duty — it's both the port the server binds to (`app.listen(PORT)`) and the port used to construct the Swagger UI server URL. This creates an impossible configuration when running behind a reverse proxy:

- The container needs `PORT=3333` to listen on a non-privileged port
- The public URL needs port `443` (or omitted entirely for HTTPS default)
- Setting `PORT=443` would crash the app trying to bind to a privileged port

### Current Behaviour

The fix in v2.0.3 (#62) replaced `req.protocol`/`req.hostname`/`req.socket.localPort` with `PROTOCOL`, `DOMAIN`, and `PORT` from config. This solved the mixed-content issue but introduced the port coupling problem.

```
# Desired: listen on 3333, Swagger URL uses https://api.example.com (no port)
# Actual: PORT=3333 → Swagger URL becomes https://api.example.com:3333
# Can't set PORT=443 → app crashes binding to privileged port
```

### Expected Behaviour

The server listen port and the public-facing URL port should be independently configurable. Setting the listen port to 3333 should not force the Swagger URL to include `:3333`.

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
Closes #66

This PR adds an `EXTERNAL_PORT` environment variable so the port used in generated URLs (Swagger UI, storage URIs) can differ from the port the server listens on. This is useful when the service runs behind a reverse proxy or is exposed on a different port via Docker port mapping (e.g. `-p 443:3333`).

## Changes
- New `buildBaseUrl` utility centralises URL construction with default port omission (443/HTTPS, 80/HTTP)
- `EXTERNAL_PORT` config variable, defaults to `PORT` for full backwards compatibility
- Startup validation rejects non-numeric `EXTERNAL_PORT` values
- All URL-generating code (Swagger, local storage, server logs) now uses `EXTERNAL_PORT` via `buildBaseUrl`
- Documentation updated in README, `.env.example`, and docs site

## Test plan
- [x] 8 unit tests for `buildBaseUrl` covering standard ports, cross-protocol edge cases, string coercion, and path appending (100% coverage)
- [x] 4 new e2e tests: EXTERNAL_PORT override, default port omission for HTTPS/443 and HTTP/80, fallback to PORT
- [x] All 64 unit tests pass
- [x] All 18 e2e tests pass
- [x] Build and lint clean

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -11,7 +12,13 @@ if (!getApiKey()) {
     process.exit(1);
 }
 
+if (process.env.EXTERNAL_PORT && isNaN(Number(process.env.EXTERNAL_PORT))) {
+    console.error(`❌ ERROR: EXTERNAL_PORT must be a valid port number, got '${process.env.EXTERNAL_PORT}'.`);
+    process.exit(1);
+}
```
[the remark this answers is not in the document]
Comment (ashleythedeveloper) [reply src/server.ts]: Done in 5444617 — now validates the resolved EXTERNAL_PORT value rather than just the env var.
